api/v1alpha1/tykoasapidefinition_types.go

```diff
@@ -0,0 +1,72 @@
+/*
+
+
+Licensed under the Mozilla Public License (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    https://www.mozilla.org/en-US/MPL/2.0/
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"github.com/TykTechnologies/tyk-operator/api/model"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+// TykOASApiDefinitionSpec defines the desired state of TykOASApiDefinition
+type TykOASApiDefinitionSpec struct {
+	// Important: Run "make" to regenerate code after modifying this file
+
+	// Foo is an example field of TykOASApiDefinition. Edit tykoasapidefinition_types.go to remove/update
+	OASRef *OASReference `json:"oas_ref"`
+}
+
+type OASReference struct {
+	model.Target `json:",inline"`
+	KeyName      string `json:"key_name"`
```

Comment:
can you please add a small comment on what this KeyName refers to?

Reply:
Fixed